include/gunrock/graph/csr.hxx

```diff
@@ -94,6 +94,65 @@ class graph_csr_t {
                                               offsets[source + 1] - 1);
   }
 
+  /**
+   * @brief Count the number of vertices belonging to the set intersection
+   * between the source and destination vertices adjacency lists. Executes a
+   * function on each intersection.
+   *
+   * @param source Index of the source vertex
+   * @param destination Index of the destination
+   * @param on_intersection Lambda function executed at each intersection
+   * @return Number of shared vertices between source and destination
+   */
+  template <typename operator_type>
+  __host__ __device__ __forceinline__ vertex_type
+  get_intersection_count(const vertex_type& source,
+                         const vertex_type& destination,
+                         operator_type on_intersection) const {
+    vertex_type intersection_count = 0;
+
+    auto intersection_source = source;
+    auto intersection_destination = destination;
+
+    auto source_neighbors_count = get_number_of_neighbors(source);
+    auto destination_neighbors_count = get_number_of_neighbors(destination);
+
+    if (source_neighbors_count > destination_neighbors_count) {
+      std::swap(intersection_source, intersection_destination);
+      std::swap(source_neighbors_count, destination_neighbors_count);
+    }
+
+    auto source_offset = offsets[intersection_source];
+    auto destination_offset = offsets[intersection_destination];
+
+    auto source_edges_iter = indices + source_offset;
+    auto destination_edges_iter = indices + destination_offset;
+
+    auto needle = *destination_edges_iter;
+    auto source_search_start =
+        search::binary::execute(source_edges_iter, needle, vertex_t{0},
+                                source_neighbors_count, search::bound_t::lower);
```

Comment:
You might consider replacing this with `thrust`. I've been debating if primitives so fundamental should have a wrapper or not, and in my implementations, I avoid my own wrapper/implementation because it is prone to errors.